soc/board_specific_workflows/test_general.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python3
+# Copyright 2021 The CFU-Playground Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import general
+import unittest
+import unittest.mock
```

Comment:
As a general principle, mocks should be used sparingly. 

Mocks are a way to open a "seam" into which one can insert tests. They are very useful especially when adding tests to code that is difficult to change. However, it's better to build the seam directly into your code if you can.

Some of the disadvantages of mocks:

1. They are verbose.
2. They add a layer of indirection and consequent complexity that makes tests more difficult to understand
3. They encourage testing of implementation-dependent interactions between rather than focussing on testing the expected functionality.

Reply:
Thanks for this, I suppose it seems obvious since you've said it but I had no idea! I've avoided mocks in my new tests, and where needed I use [autospeccing](https://docs.python.org/3/library/unittest.mock.html#auto-speccing) to be certain of a correct interface.